init.rb

```diff
@@ -65,10 +67,19 @@ def data
               },
               :fetch_timeout => 60,
               :push_timeout => 60
+            },
+            :job => {
+              :branch => choose_branch
             }
           }
         end
 
+        def choose_branch
+          return branch if branch
```

Comment:
Consequently, `branch` should not be an `attr_accessor`, but a wrapper to choose a proper value based on `@branch`.